### PR TITLE
ref(shared-cache): Counts bytes to/from shared cache

### PR DESCRIPTION
**Pull request by flub**
This adds some counters for bytes, on the metrics-side of things we
can turn that into a rough transfer rate to give us some idea of how
much we're transferring.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator/src/services/shared_cache.rs [hunk 1]
```diff
@@ -135,6 +141,7 @@ impl GcsState {
     }
 
     async fn store(&self, key: SharedCacheKey, mut src: File) -> Result<SharedCacheStoreResult> {
+        let total_bytes = src.seek(SeekFrom::End(0)).await?;
```
Comment (Swatinem) [inline crates/symbolicator/src/services/shared_cache.rs:144]: you have a `rewind()` right below which already existed before. Given that, what do we expect the current seek position is when calling this function? Is it at the beginning? (which means the previously `rewind()` was useless) Or is it undefined (which means we need to add another explicit `rewind()` here.

File: crates/symbolicator/src/services/shared_cache.rs [hunk 2]
```diff
@@ -473,17 +492,22 @@ impl SharedCacheService {
...
+            Ok(Some(bytes)) => {
                 metric!(
-                    counter("services.shared_cache.store") += 1,
+                    counter("services.shared_cache.fetch") += 1,
                     "cache" => key.name.as_ref(),
                     "hit" => "true",
                 );
+                let bytes: i64 = bytes.try_into().unwrap_or(i64::MAX);
+                metric!(
+                    counter("services.shared_cache.fetch.bytes") += bytes,
```
Comment (Swatinem) [inline crates/symbolicator/src/services/shared_cache.rs:503]: don’t we use a different metric type for these histograms?

**Review by Swatinem — APPROVED** [review]

File: crates/symbolicator/src/services/shared_cache.rs (hunk 1 above)
Comment (flub) [reply crates/symbolicator/src/services/shared_cache.rs:144]: I was treating the position before calling this as undefined, which i think is the more defensive approach.

I don't think we need a second rewind: here we explicitly go to the end to find out the size.  then explicitly go to the beginning so we can use `io::copy()`.

Comment (Swatinem) [reply crates/symbolicator/src/services/shared_cache.rs:144]: Oh, I maybe misunderstood the function signature. The function will `then yield the new position in the object`, not sure why I was thinking about it returning the number of bytes that were seeked over. So I guess this was just my misunderstanding.

File: crates/symbolicator/src/services/shared_cache.rs (hunk 2 above)
Comment (flub) [reply crates/symbolicator/src/services/shared_cache.rs:503]: (turns out i had a half-written response here... i'm good at this)

The aim here is not to get an idea of the object sizes uploaded with percentiles of their sizes etc, object sizes already have a metric.  I'm only after the rate, bytes/second, of uploading and downloading to the shared cache so we know how much bandwidth we're spending on this.  This is to verify that we're not using too much of the available machine's bandwidth and that the throttling we have in place is sufficient to not disrupt symbolicator's normal work.

**Pull request merged**